src/lib/init.js

```diff
@@ -36,49 +34,30 @@ function readExternalConsentData(config) {
 	});
 }
 
-function readInternalConsentData() {
-	return Promise.all([
-		[
-			readVendorConsentCookie(),
-			fetchPubVendorList(),
-			readPublisherConsentCookie()
-		]
-	]);
-}
-
-function readGlobalAndExternalConsentData(config) {
-	return Promise.all([
-		readExternalConsentData(config),
-		readVendorConsentCookie()
-	]);
-}
-
-export function init(configUpdates) {
+export function init (configUpdates) {
 	config.update(configUpdates);
 	log.debug('Using configuration:', config);
 
 	// Fetch the current vendor consent before initializing
-	return ((config.getConsentData) ? readGlobalAndExternalConsentData(config) : readInternalConsentData())
-		.then(([[vendorConsentData, pubVendorsList, publisherConsentData], globalVendorConsentData]) => {
-			const {vendors} = pubVendorsList || {};
+	return ((config.getConsentData) ? readExternalConsentData(config) : readConsentCookie())
+		.then((consentData) => {
+			const cmpApi = new CmpApi(CMP_ID, CMP_VERSION);
 
-			// Check config for allowedVendorIds then the pubVendorList
-			const {allowedVendorIds: configVendorIds} = config;
-			const allowedVendorIds = configVendorIds instanceof Array && configVendorIds.length ? configVendorIds :
-				vendors && vendors.map(vendor => vendor.id);
+			if (config.decoratePageCallHandler) {
+				config.decoratePageCallHandler(CmpApi, cmpApi);
+			}
 
 			// Initialize the store with all of our consent data
 			const store = new Store({
 				cmpVersion: CMP_VERSION,
 				cmpId: CMP_ID,
 				cookieVersion: COOKIE_VERSION,
-				vendorConsentData,
-				globalVendorConsentData,
-				publisherConsentData,
-				pubVendorsList,
-				allowedVendorIds
+				consentData,
+				cmpApi
 			});
 
+			cmpApi.customCommands = createCommands(store);
```

Comment:
Using `customCommands` property may be tricky. According to [documentation](https://github.com/InteractiveAdvertisingBureau/iabtcf-es/tree/master/modules/cmpapi#custom-commands) official way to add custom commands is to do this in `CmpApi` constructor. In the future implementation of `CmpApi` class may change so `customCommands` property may be no longer used. And then we may even wouldn't notice it.

Reply:
Done.